### PR TITLE
Revamp Time Cost Feature

### LINKED ISSUE #3502: Advanced costing for print jobs.
**Is your feature request related to a problem? Please describe.**
Currently the print cost shown on the preview page is (I believe to be) a simple calculation of filament cost multiplied by the amount of filament used. This provides a great estimate of material cost for the print but does not provide an accurate cost of the print job as it does not include some key costs:

- Energy
- Print time costs
- Printer depreciation
- Load / unload fixed costs

The current cost estimation provides an inaccurate measure of true printing cost.

image

**Which printers will be beneficial to this feature**
All prints. 

**Describe the solution you'd like**
Allow users to enter extra metrics about their machine for fixed costs and running time costs. Fixed costs can be used to cover printer loading and unloading / packaging / slicing / setup etc while running time cost can be used to estimate energy use, depreciation, maintenance, etc costs.

Total Estimation screen will then show the extra metrics like so:
Material Cost: 5.75
Running Cost: 17.50
Setup Cost: 3
Total Job Cost: 26.25

**Describe alternatives you've considered**
Currently this calculation is typically done by copying print time and material use figures to an external spreadsheet but this requires manual steps to extract slicer data making it slow and prone to errors. It is also unlikely that average users are completing these extra steps for standard prints leaving them without a realistic understanding of cost. 

**Additional context**

### PR DESCRIPTION
**Pull request by cochcoder**
For whatever reason GitHub decided to close the original PR which was #4392, that's why I reopened it here

EDIT: Forgot to add this: closes #3502

### CODE REVIEW

File: src/libslic3r/Fill/FillRectilinear.cpp [hunk 1]
```diff
@@ -2564,10 +2564,10 @@ static std::vector<MonotonicRegionLink> chain_monotonic_regions(
 		}
 
 		// Reinforce the path pheromones with the best path.
-        float total_cost = best_path_length + float(EPSILON);
+        float total_filament_cost = best_path_length + float(EPSILON);
```
Comment (SoftFever) [inline src/libslic3r/Fill/FillRectilinear.cpp:2567]: can you revert the changes here?

File: src/libslic3r/Fill/FillRectilinear.cpp [hunk 2]
```diff
@@ -2564,10 +2564,10 @@ static std::vector<MonotonicRegionLink> chain_monotonic_regions(
...
         for (size_t i = 0; i + 1 < path.size(); ++ i) {
             MonotonicRegionLink &link = path[i];
-            link.next->pheromone = (1.f - pheromone_evaporation) * link.next->pheromone + pheromone_evaporation / total_cost;
+            link.next->pheromone = (1.f - pheromone_evaporation) * link.next->pheromone + pheromone_evaporation / total_filament_cost;
```
Comment (SoftFever) [inline src/libslic3r/Fill/FillRectilinear.cpp:2570]: can you revert the changes here?

**Comment by SoftFever** [comment]
@cochcoder 
Thank you for your patience.

I'm not very sure about how useful it is, especially since the electricity cost largely depends on the type of filament used. The electricity consumption varies significantly between printing ABS and printing PLA. It's probably more practical to consider the cost within the filament expenses instead. It seems a bit like over-engineering to further break down the costs and clutter the UI.

Hence, I'm inclined not to include the changes.

**Comment by FillPavS** [comment]
> @cochcoder Thank you for your patience.
> 
> I'm not very sure about how useful it is, especially since the electricity cost largely depends on the type of filament used. The electricity consumption varies significantly between printing ABS and printing PLA. It's probably more practical to consider the cost within the filament expenses instead. It seems a bit like over-engineering to further break down the costs and clutter the UI.
> 
> Hence, I'm inclined not to include the changes.

Previously, there was a function for calculating the price depending on the amount of plastic used and the operating time of the printer. Now it does not take into account the operating time of the printer. The field where you can specify the price remains in the printer settings, but it does not affect anything.

**Comment by cochcoder** [comment]
> @cochcoder 
> 
> Thank you for your patience.
> 
> 
> 
> I'm not very sure about how useful it is, especially since the electricity cost largely depends on the type of filament used. The electricity consumption varies significantly between printing ABS and printing PLA. It's probably more practical to consider the cost within the filament expenses instead. It seems a bit like over-engineering to further break down the costs and clutter the UI.
> 
> 
> 
> Hence, I'm inclined not to include the changes.

I see where you're coming from, I'm going to mark this as a draft for now, as I want to look into possibly making a more accurate electricity cost calculator, and see how useful this PR would be in the future.

**Comment by cochcoder** [comment]
> > @cochcoder Thank you for your patience.
> 
> > 
> 
> > I'm not very sure about how useful it is, especially since the electricity cost largely depends on the type of filament used. The electricity consumption varies significantly between printing ABS and printing PLA. It's probably more practical to consider the cost within the filament expenses instead. It seems a bit like over-engineering to further break down the costs and clutter the UI.
> 
> > 
> 
> > Hence, I'm inclined not to include the changes.
> 
> 
> 
> Previously, there was a function for calculating the price depending on the amount of plastic used and the operating time of the printer. Now it does not take into account the operating time of the printer. The field where you can specify the price remains in the printer settings, but it does not affect anything.

I'm confused on what you mean as the cost calculator in place right now does take the time of the print into account.

**Comment by cochcoder** [comment]
I'm not having this issue, can you send a screen recording of you replicating this issue?

**Comment by FillPavS** [comment]
> I'm not having this issue, can you send a screen recording of you replicating this issue?

The built-in screen recording in Windows records only the active window. I had to record it from my phone. I hope it's clear what's going on there)
video - https://cloud.mail.ru/public/sLkn/7gRwY8NRi

**Comment by cochcoder** [comment]
@FillPavS So sorry for the late response, I've opened a new PR (#5698) to fix this issue
